scripts/base/frameworks/analyzer/logging.zeek

```diff
@@ -0,0 +1,142 @@
+##! Logging analyzer confirmations and violations into analyzer.log
+
+@load base/frameworks/config
+@load base/frameworks/logging
+
+@load ./main
+
+module Analyzer::Logging;
+
+export {
+	## Add the analyzer logging stream identifier.
+	redef enum Log::ID += { LOG };
+
+	## A default logging policy hook for the stream.
+	global log_policy: Log::PolicyHook;
+
+	## The record type defining the columns to log in the analyzer logging stream.
+	type Info: record {
+		## Timestamp of confirmation or violation.
+		ts:             time              &log;
+		## "violation" or "confirmation"
+		cause:          string            &log;
+		## "packet", "file" or "protocol"
+		analyzer_kind:  string            &log;
+		analyzer_name:  string            &log;
+		## Connection UID if available.
+		uid:            string            &log &optional;
+		## Connection identifier if available
+		id:             conn_id           &log &optional;
+		## File UID if available.
+		fuid:           string            &log &optional;
+
+		## Failure or violation reason, if available.
+		failure_reason: string            &log &optional;
+
+		## Data causing failure or violation if available.
+		## Truncated to 40 bytes.
+		failure_data:   string            &log &optional;
+	};
+
+	## Enable analyzer.log
+	option enable = T;
+
+	## Enable analyzer_confirmation, too. They are usually less
+	## interesting outside of
```

Comment:
outside of what? :-)

Reply:
Fixed and added a few more things to the whole file..